packages/relay/src/TransactionManager.ts

```diff
@@ -260,8 +299,16 @@ data                     | ${transaction.data}
     currentBlock: ShortBlockInfo,
     newMaxFee: number,
     newMaxPriorityFee: number,
-    isMaxGasPriceReached: boolean): Promise<SignedTransactionDetails> {
-    await this.validateBalance(tx.from, newMaxFee, tx.gas)
+    isMaxGasPriceReached: boolean): Promise<
+    {
+      signedTransactionDetails?: SignedTransactionDetails
+      balanceRequiredDetails: BalanceRequiredDetails
+    }> {
+    const signerBalance = await this.contractInteractor.getBalance(tx.from)
+    const balanceRequiredDetails = await this.validateBalance(tx.from, newMaxFee, tx.gas, signerBalance)
+    if (!balanceRequiredDetails.isSufficient) {
```

Comment:
Again duplicate?

Reply:
Again, here I don't throw but return a required balance. I will not adjust the replenishing amount, but only print an error if I know for sure the target replenishing amount is not enough to "unstuck" this relay.